facebook/create-react-app
# Fix argument `allowedHost`

This PR fixes issues #12304 and #11762 – “`react-scripts start` fails when both `HOST` and `proxy` are defined”.

### Steps to verify the issue

- Add a `proxy` in `package.json` of a new project (if verifying within the monorepo code, use `packages/react-scripts/package.json`).
- Set a `HOST` environment variable and run the `start` script, e.g: `HOST=example.local yarn run start`
- The script will fail with error message: `Invalid options object. Dev Server has been initialized using an options object that does not match the API schema. options.allowedHosts[0] should be a non-empty string.`

After applying this patch, the script will run as expected, setting the `allowedHost` property of the Webpack config to the given hostname.

### Explanation

The function [`createDevServerConfig`](https://github.com/facebook/create-react-app/blob/d960b9e38c062584ff6cfb1a70e1512509a966e7/packages/react-scripts/config/webpackDevServer.config.js#L24) expects the hostname as string for its second argument, but instead of `HOST`, `urls.lanUrlForConfig` is provided and its value is `undefined`.

It appears, this bug has been in introduced all the way back in [2017](https://github.com/facebook/create-react-app/commit/8dc9217b61d8a2464fc6e5fdd5b1af925d572e02).
